### PR TITLE
fix(ingest/dbt): fix dbt catalog version check

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py [hunk 1]
```diff
@@ -680,10 +695,10 @@ def get_custom_properties(node: DBTNode) -> Dict[str, str]:
 
 
 def _get_dbt_cte_names(name: str, target_platform: str) -> List[str]:
-    # Match the dbt CTE naming scheme:
-    # The default is defined here https://github.com/dbt-labs/dbt-core/blob/4122f6c308c88be4a24c1ea490802239a4c1abb8/core/dbt/adapters/base/relation.py#L222
-    # However, since this PR https://github.com/dbt-labs/dbt-core/pull/2712, it's also possible
-    # for adapters to override this default. Only a handful actually do though:
+    # Match the dbt CTE naming scheme: The default is defined here
+    # https://github.com/dbt-labs/dbt-core/blob/4122f6c308c88be4a24c1ea490802239a4c1abb8/core/dbt/adapters/base
+    # /relation.py#L222 However, since this PR https://github.com/dbt-labs/dbt-core/pull/2712, it's also possible for
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py:700]: undo this comment change - can't be breaking up links

overall - why are all these comments and strings being reformatted?

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py [hunk 2]
```diff
@@ -561,7 +565,7 @@ def load_nodes(self) -> Tuple[List[DBTNode], Dict[str, Optional[str]]]:
         if (
             catalog_version
             and catalog_version.startswith("1.7.")
-            and catalog_version < "1.7.3"
+            and version.parse(catalog_version) < version.parse("1.7.3")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py:568]: add some exception handling around this, just in case

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py:700]: they are going over the 80 characters limit and hard to follow as long line. I will revert this

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py (hunk 2 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py]: done

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py]: reverted the changes.

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py [hunk 3]
```diff
@@ -558,16 +562,26 @@ def load_nodes(self) -> Tuple[List[DBTNode], Dict[str, Optional[str]]]:
...
+                    title="Dbt Catalog Version",
+                    message="Due to a bug in dbt version between 1.7.0 and 1.7.2, you will have incomplete metadata "
+                    "source",
+                    context=f"Due to a bug in dbt, dbt version {catalog_version} will have incomplete metadata on "
+                    f"sources."
+                    "Please upgrade to dbt version 1.7.3 or later. "
+                    "See https://github.com/dbt-labs/dbt-core/issues/9119 for details on the bug.",
+                )
+        except Exception as e:
             self.report.report_warning(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py:581]:
```suggestion
            self.report.report_info(
```

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py [hunk 4]
```diff
@@ -558,16 +562,26 @@ def load_nodes(self) -> Tuple[List[DBTNode], Dict[str, Optional[str]]]:
...
+                    title="Dbt Catalog Version",
+                    message="Due to a bug in dbt version between 1.7.0 and 1.7.2, you will have incomplete metadata "
+                    "source",
+                    context=f"Due to a bug in dbt, dbt version {catalog_version} will have incomplete metadata on "
+                    f"sources."
+                    "Please upgrade to dbt version 1.7.3 or later. "
+                    "See https://github.com/dbt-labs/dbt-core/issues/9119 for details on the bug.",
+                )
+        except Exception as e:
+            self.report.report_info(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_core.py:581]:
```suggestion
            self.report.info(
```

**Review by hsheth2 — APPROVED** [review]
